### DISCUSSION TITLE
Oauth 1 and  Oauth 2

### DISCUSSION THREAD

**Discussion by jason18505** [danieldevine/bird-elephant#46] [Q&A] [answered]
From my understanding media uploads are only supported in v1 with Oauth 1.

With your library will I need get users to go through both Oauth flows so I can upload media with Oauth 1 but also call "$tweets->get()" which requires Oauth 2 bearer tokens per your documentation?

**Comment by danieldevine** [comment, marked as answer]
Short answer is no. The bearer token would be supplied by your app rather than the user so if the user is logged in via an OAuth 1.0 flow everything will be fine.
